pages/account/statistics.js

```diff
@@ -79,6 +79,8 @@ export async function getServerSideProps(context) {
     profile.links.some((pLink) => pLink.url === link.url)
   );
 
+  const dailyStatsMap = new Map(data.profile.daily.map(stat => [stat.date, stat]));
```

Comment:
stat.date is a `dateTime` object with UTC. 

It will not match your condition i.e `if (!dailyStatsMap.has(date))` becaue date here is in "YYYY-MM-DD" format.
You already splited `return date.toISOString().split('T')[0]; // Get date in "YYYY-MM-DD" format`

Solution: Create `dailyStatsMap` constructor:
`const dailyStatsMap = new Map(data.profile.daily.map(stat => [new Date(stat.date).toISOString().split('T')[0], stat]));
`

Reply:
please check